doctrine/events.rst

```diff
@@ -0,0 +1,345 @@
+
+    This article covers listeners and subscribers for Doctrine ORM. If you are
+    using ODM for MongoDB, read the `DoctrineMongoDBBundle documentation`_.
+
+Doctrine Lifecycle Callbacks
+----------------------------
+
+Lifecycle callbacks are defined as methods inside the entity you want to modify.
+For example, suppose you want to set a ``createdAt`` date column to the current
+date, but only when the entity is first persisted (i.e. inserted). To do so,
+define a callback for the ``prePersist`` Doctrine event:
+
+.. configuration-block::
+
+    .. code-block:: php-annotations
+
+        // src/Entity/Product.php
+        use Doctrine\ORM\Mapping as ORM;
+
+        // When using annotations, don't forget to add @ORM\HasLifecycleCallbacks()
+        // to the class of the entity where you define the callback
+
+        /**
+         * @ORM\Entity()
+         * @ORM\HasLifecycleCallbacks()
+         */
+        class Product
+        {
+            // ...
+
+            /**
+             * @ORM\PrePersist
+             */
+            public function setCreatedAtValue()
+            {
+                $this->createdAt = new \DateTime();
+            }
+        }
+
+    .. code-block:: yaml
+
+        # config/doctrine/Product.orm.yml
+        App\Entity\Product:
+            type: entity
+            # ...
+            lifecycleCallbacks:
+                prePersist: ['setCreatedAtValue']
+
+    .. code-block:: xml
+
+        <!-- config/doctrine/Product.orm.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping
+                https://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">
+
+            <entity name="App\Entity\Product">
+                <!-- ... -->
+                <lifecycle-callbacks>
+                    <lifecycle-callback type="prePersist" method="setCreatedAtValue"/>
+                </lifecycle-callbacks>
+            </entity>
+        </doctrine-mapping>
+
+.. note::
+
+    Some lifecycle callbacks receive an argument that provides access to
+    useful information such as the current entity manager (e.g. the ``preUpdate``
+    callback receives a ``PreUpdateEventArgs $event`` argument).
+
+.. _doctrine-listener:
+
+Doctrine Lifecycle Listeners
+----------------------------
+
+Lifecycle listeners are defined as PHP classes that listen to a single Doctrine
+event on all the application entities. For example, suppose that you want to
+update some search index whenever a new entity is persisted in the database. To
+do so, define a listener for the ``postPersist`` Doctrine event::
+
+    // src/EventListener/SearchIndexer.php
+    namespace App\EventListener;
+
+    use App\Entity\Product;
+    // for Doctrine < 2.4: use Doctrine\ORM\Event\LifecycleEventArgs;
```

Comment:
I suggest dropping that comment. Doctrine 2.3 is dead since years, and not supported by DoctrineBundle anymore anyway.